components/kyma-environment-broker/internal/cis/client.go

```diff
@@ -0,0 +1,156 @@
+package cis
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+const (
+	eventServicePath = "%s/events/v1/events/central"
+	eventType        = "Subaccount_Deletion"
+	defaultPageSize  = "150"
+)
+
+type Config struct {
+	ClientID        string
+	ClientSecret    string
+	AuthURL         string
+	EventServiceURL string
+	PageSize        string `envconfig:"optional"`
+}
+
+type Client struct {
+	httpClient *http.Client
+	config     Config
+	log        logrus.FieldLogger
+}
+
+func NewClient(ctx context.Context, config Config, log logrus.FieldLogger) *Client {
+	cfg := clientcredentials.Config{
+		ClientID:     config.ClientID,
+		ClientSecret: config.ClientSecret,
+		TokenURL:     config.AuthURL,
+	}
+	httpClientOAuth := cfg.Client(ctx)
+	httpClientOAuth.Timeout = 30 * time.Second
+
+	if config.PageSize == "" {
+		config.PageSize = defaultPageSize
+	}
+
+	return &Client{
+		httpClient: httpClientOAuth,
+		config:     config,
+		log:        log.WithField("client", "CIS-2.0"),
+	}
+}
+
+// SetHttpClient auxiliary method of testing to get rid of oAuth client wrapper
+func (c *Client) SetHttpClient(httpClient *http.Client) {
+	c.httpClient = httpClient
+}
+
+type subAccounts struct {
+	total int
+	ids   []string
+	from  time.Time
+	to    time.Time
+}
+
+func (c *Client) FetchSubAccountsToDelete() ([]string, error) {
+	subaccounts := subAccounts{}
+
+	err := c.fetchSubAccountsFromDeleteEvents(&subaccounts, 0)
+	if err != nil {
+		return []string{}, errors.Wrap(err, "while fetching subaccounts from delete events")
+	}
+
+	c.log.Infof("CIS returned total amount of delete events: %d, client fetched %d subaccounts to delete. "+
+		"The events includes a range of time from %s to %s",
+		subaccounts.total,
+		len(subaccounts.ids),
+		subaccounts.from,
+		subaccounts.to)
+
+	return subaccounts.ids, nil
+}
+
+func (c *Client) fetchSubAccountsFromDeleteEvents(collection *subAccounts, page int) error {
+	request, err := c.buildRequest(page)
+	if err != nil {
+		return errors.Wrap(err, "while building request for event service")
+	}
+
+	response, err := c.httpClient.Do(request)
+	if err != nil {
+		return errors.Wrap(err, "while executing request to event service")
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return fmt.Errorf("while processing response: %s", c.handleWrongStatusCode(response))
+	}
+
+	var cisResponse CisResponse
+	err = json.NewDecoder(response.Body).Decode(&cisResponse)
+	if err != nil {
+		return errors.Wrap(err, "while decoding CIS response")
+	}
+
+	collection.total = cisResponse.Total
+	for _, event := range cisResponse.Events {
+		if event.Type != eventType {
```

Comment:
theoretically it guaranties but I don't think it would hurt to double-check. If query filter wouldn't work then we can remove instance from subaccount which wasn't deleted.